oar/core/shipment.py

```diff
@@ -0,0 +1,236 @@
+            # Try to get merge request directly by ID first
+            try:
+                self.mr = project.mergerequests.get(merge_request_id)
+            except gitlab.exceptions.GitlabGetError:
+                # Fall back to listing and filtering if direct get fails
+                mrs = project.mergerequests.list()
+                self.mr = next((mr for mr in mrs if mr.iid == merge_request_id), None)
+                if not self.mr:
+                    raise Exception(f"Merge request {merge_request_id} not found")
+        except gitlab.exceptions.GitlabGetError as e:
+            raise Exception(f"Failed to get merge request: {str(e)}")
+
+    def get_file_content(self, file_path: str) -> str:
+        """Get raw file content from the merge request"""
+        if not file_path or not isinstance(file_path, str):
+            raise ValueError("File path must be a non-empty string")
+            
+        try:
+            file_content = self.gl.projects.get(self.project_name).files.get(
+                file_path=file_path,
+                ref=self.mr.source_branch
+            )
+            if isinstance(file_content, str):
+                return file_content
+            return file_content.decode().decode("utf-8")
+        except Exception as e:
+            raise Exception(f"Failed to access file '{file_path}': {str(e)}")
+
+    def get_all_files(self, file_extension: str = None) -> List[str]:
+        """Get all files changed in the merge request, optionally filtered by extension
+        
+        Args:
+            file_extension: Optional file extension to filter by (e.g. 'yaml')
+            
+        Returns:
+            List of file paths changed in the merge request
+        """
+        try:
+            project = self.gl.projects.get(self.project_name)
+            changes =  self.mr.changes()
+            all_files = [change['new_path'] for change in changes.get('changes', [])]
+            
+            if file_extension:
+                return [f for f in all_files if f.lower().endswith(f'.{file_extension.lower()}')]
+            return all_files
+        except Exception as e:
+            raise Exception(f"Failed to get changed files: {str(e)}")
+
+    def add_comment(self, comment: str) -> None:
+        """Add comment to the merge request"""
+        try:
+            self.mr.notes.create({'body': comment})
+        except Exception as e:
+            raise Exception(f"Failed to add comment: {e}")
+
+
+class ShipmentData:
+    """Class for handling shipment merge request operations"""
+    
+    def __init__(self, config_store: ConfigStore):
+        """Initialize with ConfigStore instance"""
+        self._cs = config_store
+        self._mrs = self._initialize_mrs()
+        
+    def _initialize_mrs(self) -> List[GitLabMergeRequest]:
+        """Initialize GitLabMergeRequest objects from shipment MRs"""
+        mrs = []
+        mr_urls = self._cs.get_shipment_mrs() or []
```

Comment:
```suggestion
        mr_urls = self._cs.get_shipment_mrs()
```
`or []` is not necessary as the `get_shipment_mrs` can returns also `[]`

Reply:
done